src/NodeConnectionInteraction.cpp

```diff
@@ -55,11 +72,56 @@ tryConnect() const
 {
   // 1) Check conditions from 'canConnect'
   PortIndex portIndex = INVALID;
+  bool typeConversionNeeded = false; 
+  std::unique_ptr<NodeDataModel> typeConverterModel;
 
-  if (!canConnect(portIndex))
+  if (!canConnect(portIndex, typeConversionNeeded, typeConverterModel))
   {
     return false;
   }
+  
+  /// 1.5) If the connection is possible but a type conversion is needed, add a converter node to the scene, and connect it properly
+  if (typeConversionNeeded)
+  {
+    //Determining port types
+    PortType requiredPort = connectionRequiredPort();
+    PortType connectedPort = requiredPort == PortType::Out ? PortType::In : PortType::Out;
+
+    //Get the node and port from where the connection starts
+    auto outNode = _connection->getNode(connectedPort);
+    auto outNodePortIndex = _connection->getPortIndex(connectedPort);
+
+    //Creating the converter node
+    Node& converterNode = _scene->createNode(std::move(typeConverterModel));
+    
+    //Calculating the converter nodes position in the scene. It'll be positioned half way between the two ports that it connects. 
+    //The first line calculates the halfway point between the ports (node position + port position on the node for both nodes averaged).
+    //The second line offsets this coordinate with the size of the converter node, so that the converter nodes center falls on the originally
+    //calculated coordinate, instead of it's upper left corner.
+    auto converterNodePos = (outNode->nodeGraphicsObject().pos() + outNode->nodeGeometry().portScenePosition(outNodePortIndex, connectedPort) +
+      _node->nodeGraphicsObject().pos() + _node->nodeGeometry().portScenePosition(portIndex, requiredPort)) / 2.0f;
+    converterNodePos.setX(converterNodePos.x() - converterNode.nodeGeometry().width() / 2.0f);
+    converterNodePos.setY(converterNodePos.y() - converterNode.nodeGeometry().height() / 2.0f);
+    converterNode.nodeGraphicsObject().setPos(converterNodePos);
```

Comment:
Move this calculation to NodeGeometry as a static function, so this can be reused in the future if a node is needed to be positioned between to other.